### DISCUSSION TITLE
postUpgradeTasks - Where to set it?

### DISCUSSION THREAD

**Discussion by dtroup-global** [renovatebot/renovate#14071] [Request Help] [answered]
Hi, general question as I can't find this in the documentation. It states in the config that you have a global option which must be set to run the commands in "postUpgradeTasks", which is fine (allowedPostUpgradeCommands). But it doesn't mention WHERE you set "postUpgradeTasks", for it to run. I've tried adding it to the global config as I want it to run on all of the repo's but it doesn't seem to be doing anything.

Rather than just a snippet of code it would be useful to see how it's used in an example. Is it possible to set this in a global setting, or can it only be set per repo for instance?

The code I'm using is:

```
{
  "allowedPostUpgradeCommands": [
    "^tf-docs.sh"
  ],
  "force": {
    "extends": [
      ":prHourlyLimitNone",
      ":semanticPrefixFix"
    ],
    "packageRules": [
      {
        "matchDatasources": [
          "terraform-module",
          "terraform-provider"
        ],
        "postUpgradeTasks": [
          {
            "commands": [
              "/tf-docs/tf-docs.sh"
            ],
            "executionMode": "branch",
            "fileFilters": [
              "README.md",
              "**/*.tf"
            ]
          }
        ]
      }
    ],
    "reviewersFromCodeOwners": true
  },
  "onboardingConfigFileName": ".github/renovate.json",
  "prCommitsPerRunLimit": 20
}
```

I'm new to this application but any help would be much appreciated

Thanks,

D

**Comment by rarkins** [comment, marked as answer]
Are you a self-hosted user? If so how are you configuring our global config? Normally we recommend `config.js`.

Also as you're admitted new, I recommend you remove the `force` object - it's almost certainly a bad idea.

**Reply by dtroup-global** [reply]
Thanks @rarkins Yes, we're self hosting. Although I'm a new user, the business is not, the force thing was there before I started looking at this I'm just attempting to add the post upgrade task to save some headaches on the PR's.

In the kubernetes example [here](https://docs.renovatebot.com/examples/self-hosting/#kubernetes)  it shows to use config.json
```
- name: RENOVATE_CONFIG_FILE
  value: '/opt/renovate/config.json'
```
My understanding that the json in my original post IS the global config

Thanks for the speedy response.

D

**Reply by rarkins** [reply]
Well, that otherwise looks ok, assuming you're not overriding those tasks somewhere else. Note though that in a regex `.` means "any character", however that includes the `.` character too so I don't think that's the problem.

You'll need to elaborate more on " it doesn't seem to be doing anything"

**Reply by dtroup-global** [reply]
The job runs, it creates the branches and PR's with the relevant configs in them, however, I also want to update the readme, which is what the tf-docs command is in the post upgrade tasks, but it doesn't run the script, there is no error and nothing to really state that anything is wrong.

D

**Reply by rarkins** [reply]
Hard to guess without debug logs for that repo and branch

**Reply by dtroup-global** [reply]
I found the issue, the commands I'm running don't seem to be able to run in the global config, so I've moved them to a shareable config instead, which seems to be working.

**Comment by viceice** [comment]
allowedPostUpgradeCommands regex doesn't match you command as it starts with a different char. 😉
